app/models/preserved_copy.rb

```diff
@@ -31,6 +31,13 @@ class PreservedCopy < ApplicationRecord
   validates :status, inclusion: { in: statuses.keys }
   validates :version, presence: true
 
+  scope :least_recent_version_audit, lambda { |last_checked_b4_date, storage_dir|
```

Comment:
😆 _rimshot_

since it's a lambda, i think you could have it check at the top of the block and throw an exception upon encountering a bad date, since what the scope is returning is the result of the AR query method chain (which it would still return if the formatting guard clause on the block failed to raise an error).

but i have no idea whether that's an idiomatic/common/acceptable way to do that sort of validation.  so i'd be inclined to have the caller do the check.

Reply:
@tallenaz i would add the suggestion to the "validate date arg" ticket, not include the work here ...